runtime/onert/api/python/package/infer/session.py

```diff
@@ -6,14 +6,35 @@ class session(BaseSession):
     """
     Class for inference using nnfw_session.
     """
-    def __init__(self, nnpackage_path, backends="cpu"):
+    def __init__(self, nnpackage_path: str = None, backends: str = "cpu"):
         """
         Initialize the inference session.
         Args:
             nnpackage_path (str): Path to the nnpackage file or directory.
             backends (str): Backends to use, default is "cpu".
         """
-        super().__init__(libnnfw_api_pybind.infer.nnfw_session(nnpackage_path, backends))
+        if nnpackage_path is not None:
+            super().__init__(
+                libnnfw_api_pybind.infer.nnfw_session(nnpackage_path, backends))
+            self.session.prepare()
+            self.set_outputs(self.session.output_size())
+        else:
+            super().__init__()
+
+    def compile(self, nnpackage_path: str, backends: str = "cpu"):
```

Comment:
Is there a reason to allow `compile(..)` interface? 
At first glance, its' role is almost the same with the constructor.

Reply:
> Is there a reason to allow compile(..) interface?

This is the history https://github.com/Samsung/ONE/pull/14525#issuecomment-2572957523

> At first glance, its' role is almost the same with the constructor.

Right. It's @hseok-oh 's suggestion.